src/test/test_proj_data_info_subsets.cxx

```diff
@@ -0,0 +1,531 @@
+    const shared_ptr<const ProjDataInfo>& template_projdatainfo_sptr,
+    const shared_ptr<const VoxelsOnCartesianGrid<float> >& template_image_sptr,
+    bool use_z_symmetries, bool use_other_symmetries)
+{
+  cerr << "\tSetting up default projector pair, ProjectorByBinPairUsingProjMatrixByBin" << endl;
+  auto proj_matrix_sptr = std::make_shared<ProjMatrixByBinUsingRayTracing>();
+  proj_matrix_sptr->set_do_symmetry_180degrees_min_phi(use_other_symmetries);
+  proj_matrix_sptr->set_do_symmetry_90degrees_min_phi(use_other_symmetries);
+  proj_matrix_sptr->set_do_symmetry_shift_z(use_z_symmetries);
+  proj_matrix_sptr->set_do_symmetry_swap_s(use_other_symmetries);
+  proj_matrix_sptr->set_do_symmetry_swap_segment(use_other_symmetries);
+  auto proj_pair_sptr = std::make_shared<ProjectorByBinPairUsingProjMatrixByBin>(proj_matrix_sptr);
+
+  proj_pair_sptr->set_up(template_projdatainfo_sptr, template_image_sptr);
+  return proj_pair_sptr;
+}
+
+
+void TestProjDataInfoSubsets::fill_proj_data_with_forward_projection(
+  const std::shared_ptr<ProjData>& proj_data_sptr,
+  const std::shared_ptr<const VoxelsOnCartesianGrid<float> >& test_image_sptr)
+{
+  cerr << "\tFilling ProjData with forward projection" << endl;
+  auto forward_projector = construct_projector_pair(
+    proj_data_sptr->get_proj_data_info_sptr(), test_image_sptr)
+      ->get_forward_projector_sptr();
+
+  forward_projector->set_input(*test_image_sptr);
+  forward_projector->forward_project(*proj_data_sptr);
+}
+
+    
+void
+TestProjDataInfoSubsets::
+run_tests()
+{
+  cerr << "-------- Testing ProjDataInfoCylindricalArcCorr --------\n";
+  try
+  {
+    // make an image (TODO: or open?)
+
+    // Open sinogram
+    if (_sinogram_filename.empty()) {
+      _input_sino_sptr = construct_test_proj_data();
+      _test_image_sptr = construct_test_image_data(*_input_sino_sptr->get_proj_data_info_sptr());
+      fill_proj_data_with_forward_projection(_input_sino_sptr, _test_image_sptr);
+    }
+    else {
+      _input_sino_sptr = ProjData::read_from_file(_sinogram_filename);
+      _test_image_sptr = construct_test_image_data(*_input_sino_sptr->get_proj_data_info_sptr());
+    }
+
+    test_split(*_input_sino_sptr);
+
+    // test_split_and_combine(*_input_sino_sptr);
+
+    test_forward_projection_is_consistent(
+      _test_image_sptr, _input_sino_sptr);
+    cerr << "repeat with an 'unusual' number of subsets, 13" << endl;
+    test_forward_projection_is_consistent(
+      _test_image_sptr, _input_sino_sptr, /*use_z_symmetries=*/false, /*use_z_symmetries=*/false, /*num_subsets=*/13);
+    cerr << "repeat with z shift symmetries" << endl;
+    test_forward_projection_is_consistent(
+      _test_image_sptr, _input_sino_sptr, /*use_z_symmetries=*/true, /*use_z_symmetries=*/false);
+    cerr << "repeat with all symmetries" << endl;
+    test_forward_projection_is_consistent(
+      _test_image_sptr, _input_sino_sptr, /*use_z_symmetries=*/true, /*use_z_symmetries=*/true);
```

Comment:
Not sure if we expected this one to run? It passes locally. Of course, some of the symmetries are automatically disabled later.